### ISSUE TITLE
required error on md-select not prevent the submit event

### ISSUE THREAD

**Issue by leibale** [angular/material#8599]
i have a form with ng-submit attribute and two inputs:
- md-input
- md-select

both of them are required
if the md-input is empty the submit is stopped (when you click the focus is set to the required input and a message pops on top of it)
if the md-select is empty the submit doesn't recognize it and goes through.

[code example](http://plnkr.co/edit/fk0mFXN1OchtmFvuKZ4y?p=preview)

**Angular Versions**:
-  Angular Version: 1.5.5
-  Angular Material Version: 1.0.9

**Comment by crisbeto** [comment]
I believe the select works as intended. Your form doesn't submit when the text field is empty, because of the default browser `required` attribute, not any of Angular's validations. If you add `novalidate` to the form, in order to disable the native browser validation, and you comment out the select, you can see that you can still submit the form with an empty input.

**Comment by leibale** [comment]
so i must add `ng-disabled` on the submit button..?
and what about required error style (red border) for `md-select`?

**Comment by crisbeto** [comment]
You can either add `ng-disabled` on the button, or pass the `statusForm.$valid` to the submit method and prevent it from submitting there. 

As for the red border, it will show up when the select is "dirty", e.g. if the user selected and then removed the value. [You can see what that looks like in the "Validations" demo at the bottom.](https://material.angularjs.org/HEAD/demo/select)

**Comment by leibale** [comment]
@crisbeto the css rule for the error message in `md-input` is `.ng-invalid` and for `md-select` is `.ng-invalid.ng-dirty`, why there is a difference?

**Comment by crisbeto** [comment]
I'm not sure really, it might have to do with UX, so an empty select doesn't get highlighted the first time the user opens the page.

**Comment by leibale** [comment]
@crisbeto sorry, the css rule for the error message in md-input is `md-input-invalid` not `.ng-invalid`. this class added only after focus or submit (and not when the page opens).
could you ask the UX team or send them a link to this issue?

**Comment by crisbeto** [comment]
I'm pretty sure that this is the intended behaviour. If you want to make it underline when the form is submitted, you can add this to your CSS:

```css
.ng-submitted md-select.ng-invalid .md-select-value{
  color: rgb(221,44,0);
  border-bottom-color: rgb(221,44,0);
}
```

**Comment by leibale** [comment]
@crisbeto there is no way that this is what the UX team wants.. if you can ask them and make sure that this is the behaviour it will be great :)

for now i will use the css you gave me

**Comment by topherfangio** [comment]
Part of this issue appears related to #8529 which I am working on today.

**Comment by leibale** [comment]
@crisbeto told you.. ;)
@topherfangio tnx :)

**Comment by adibas03** [comment]
Issue Fixed in version 1.13 thanks @topherfangio
